src/main/webapp/resources/js/pages/user/components/UserDetailsPage.jsx

```diff
@@ -0,0 +1,196 @@
+import React, { useState } from "react";
+import { useParams } from "react-router-dom";
+import {
+  Button,
+  Form,
+  Input,
+  notification,
+  Select,
+  Space,
+  Switch,
+  Typography,
+} from "antd";
+import { formatDate } from "../../../utilities/date-utilities";
+
+import {
+  useGetUserDetailsQuery,
+  useEditUserDetailsMutation,
+} from "../../../apis/users/users";
+
+/**
+ * React component to display the user details page.
+ * @returns {*}
+ * @constructor
+ */
+export default function UserDetailsPage() {
+  const { userId } = useParams();
+  const { data, isSuccess } = useGetUserDetailsQuery(userId);
+  const [editUser] = useEditUserDetailsMutation();
+
+  const [formErrors, setFormErrors] = useState();
+
+  const onFormFinish = (values) => {
+    editUser({ userId: userId, ...values })
+      .unwrap()
+      .then((payload) => {
+        notification.success({
+          message: i18n("UserDetailsPage.notification.success"),
+        });
+      })
+      .catch((error) => {
+        notification.error({
+          message: i18n("UserDetailsPage.notification.error"),
+        });
+        setFormErrors(error.data);
+      });
+  };
+
+  if (isSuccess) {
+    return (
+      <Space direction="vertical">
+        <Typography.Title level={4}>{data.user.username}</Typography.Title>
+        <Form
+          layout="vertical"
+          initialValues={data.user}
+          onFinish={onFormFinish}
+        >
+          <Form.Item
+            label={i18n("UserDetailsPage.form.label.firstName")}
+            name="firstName"
+            help={
+              formErrors?.find((error) => error.field === "firstName")?.message
```

Comment:
Create a function where you just pass in the field name and then call that instead of having 

` formErrors?.find((error) => error.field === "firstName")?.message`
` formErrors?.find((error) => error.field === "firstName")`
..etc for each of the fields. 

You can pass in an extra parameter which can just be a true or false value of whether you want the message returned for the field or just a boolean value. In the function you would just check for this boolean value and return what is necessary.

This is what the function would look like:
`const getError = (fieldName, message) => {};`

If message is true then return the message for that field otherwise return if there is an error or not for the field.

Then you would just have this for each of the fields:

`help={ getError("firstName", true) }`
`validateStatus={ getError("firstName", false) }`

Hope that makes sense :)

Reply:
I like the map for storing and retrieving errors. Please see 560e696bd8afa4fdca973bb9e2518b39e4733a61.